Tensile/kernelConverter.py

```diff
@@ -0,0 +1,317 @@
+    
+  MatrixM,MatrixN,TT0,TT1,MT0,MT1,WG0,WG1,Waves = calculateMatrixMNThreadTileMacroTileWorkGroupParameters(MIBlock,MIWaveTile,MIWaveGroup)
+ 
+  if not currentSolution.keys() & {'MacroTile0','MacroTile1','ThreadTile','WorkGroup','MatrixInstM','MatrixInstN'}:
+    raise RuntimeError("one or more of these fields =>MacroTile0, MacroTile1, ThreadTile, WorkGroup,MatrixInstM,MatrixInstN is missing in the library logic file!! ..\n")
+   
+  if MT0 != currentSolution["MacroTile0"]:
+   raise RuntimeError("Macro Tile0 {0} doesnot match LibLogic value {1}".format(MT0, currentSolution["MacroTile0"]))
+  
+  if MT1 !=  currentSolution["MacroTile1"]:
+   raise RuntimeError("Macro Tile1 {0} doesnot match LibLogic value {1}".format(MT1, currentSolution["MacroTile1"]))
+    
+  if TT0 != int(currentSolution["ThreadTile"][0]):
+   raise RuntimeError("ThreadTile0 {0} doesnot match LibLogic value {1}".format(TT0, currentSolution["ThreadTile"][0]))
+   
+  #print1("ThreadTile0 {0} matches LibLogic value {1}\n".format(TT0, currentSolution["ThreadTile"][0])) 
+    
+  if TT1 !=  int(currentSolution["ThreadTile"][1]):
+   raise RuntimeError("ThreadTile1 {0} doesnot match LibLogic value {1}".format(TT1, currentSolution["ThreadTile"][1]))
+   
+  #print1("ThreadTile0 {1} matches LibLogic value {1}\n".format(TT1, currentSolution["ThreadTile"][1]))
+    
+  if WG0 !=  int(currentSolution["WorkGroup"][0]):
+   raise RuntimeError("WorkGroup0 {0} doesnot match LibLogic value {1}\n".format(WG0, currentSolution["WorkGroup"][0]))
+  
+  #print1("WorkGroup0 {0} matches LibLogic value {1}".format(WG0, currentSolution["WorkGroup"][0]))
+    
+  if WG1 !=  int(currentSolution["WorkGroup"][1]):
+   raise RuntimeError("WorkGroup1 {1} doesnot match LibLogic value {1}".format(WG1, currentSolution["WorkGroup"][1]))
+  
+  #print1("WorkGroup1 {1} does matches LibLogic value {1}\n".format(WG1, currentSolution["WorkGroup"][1]))
+  
+  if MatrixM !=  int(currentSolution["MatrixInstM"]):
+   raise RuntimeError("MatrixInstM {0} doesnot match LibLogic value {1}".format(MatrixM, currentSolution["MatrixInstM"]))
+  
+  if MatrixN !=  int(currentSolution["MatrixInstN"]):
+   raise RuntimeError("MatrixInstN {0} doesnot match LibLogic value {1}".format(MatrixN, currentSolution["MatrixInstN"]))
+
+  #print1("Thread Tile, Macro Tile ,WG , Matrix M, Matrix N parameter matches from the lib logic file\n")
+
+def calculateMatrixMNThreadTileMacroTileWorkGroupParameters(MIBlock,MIWaveTile,MIWaveGroup):
+
+  MatrixM = int(MIBlock[0]) *int(MIBlock[4])
+  MatrixN = int(MIBlock[1])
+  TT0 = int(MIWaveTile[0]) 
+  TT1 = int(MIWaveTile[1])*int(MIBlock[1])
+  MT0 =  MatrixM*int(MIWaveTile[0])*int(MIWaveGroup[0])
+  MT1 =  MatrixN*int(MIWaveTile[1])*int(MIWaveGroup[1])
+  WG0 =  int(MIBlock[0]) * int(MIBlock[4])*int(MIWaveGroup[0])
+  WG1 = int(MIWaveGroup[0]) * int(MIWaveGroup[1]) *64 // int(WG0)

```

Comment:
Don't we need to use WavefrontSize instead of 64?

Reply:
some kernel does not have WavefrontSize parameter,  in that case , do we have to default to 64?

Reply:
Like other parameters, we should be able to use the default value if it is not explicitly defined.